i6_models/parts/conformer/mhsa.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Optional, Callable
+import torch
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class ConformerMHSAV1Config(ModelConfiguration):
+    input_dim: int  # input dim and total dimension for query/key and value projections, should be dividable by `num_att_heads`
```

Comment:
In [torch](https://pytorch.org/docs/stable/_modules/torch/nn/modules/activation.html#MultiheadAttention) they hard check for this divisibilty with 
`self.head_dim = embed_dim // num_heads`
`assert self.head_dim * num_heads == self.embed_dim, "embed_dim must be divisible by num_heads"`.
I think it would be nice to have this check in the config already, this should be doable through the `post_init` similar to typing. 
Maybe we could extend this even further and introduce something like `value_check()` which in the base config is just a `pass`, but can be overwritten in each config (if there are checks to make ofc) and then get automatically called by the `__post_init` of the `ModelConfiguration` base class. 

Even though this is resolved now, in the Conformer part this `assert (kernel_size - 1) % 2 == 0`  could have been resolved as well (I know its not required anymore just as an example). This way we can keep the error where the values are actually set and since the config is closely tied to the module this should work in my opinion. 

Any opinions?

Reply:
i think this kind of check will be quite common, so adding this value_check() sounds quite reasonable.

Reply:
C.f. #12